signer/tests/integration/block_observer.rs

```diff
@@ -1415,3 +1420,251 @@ async fn block_observer_updates_dkg_shares_after_observing_bitcoin_block() {
+
+    let mut signal_receiver = ctx.get_signal_receiver();
+
+    // The block observer reaches out to the stacks node to get the most
+    // up-to-date information. We don't have stacks-core running so we mock
+    // these calls.
+    ctx.with_stacks_client(|client| {
+        client
+            .expect_get_tenure_info()
+            .returning(move || Box::pin(std::future::ready(Ok(DUMMY_TENURE_INFO.clone()))));
+
+        let chain_tip = BitcoinBlockHash::from(chain_tip_info.hash);
+        client.expect_get_tenure().returning(move |_| {
+            let mut tenure = TenureBlocks::nearly_empty().unwrap();
+            tenure.anchor_block_hash = chain_tip;
+            Box::pin(std::future::ready(Ok(tenure)))
+        });
+
+        client.expect_get_pox_info().returning(|| {
+            let response = serde_json::from_str::<RPCPoxInfoData>(GET_POX_INFO_JSON)
+                .map_err(Error::JsonSerialize);
+            Box::pin(std::future::ready(response))
+        });
+    })
+    .await;
+
+    // ** Step 2 **
+    //
+    // Start the BlockObserver
+    //
+    // We only proceed with the test after the process has started, and
+    // we use this counter to notify us when that happens.
+    let start_flag = Arc::new(AtomicBool::new(false));
+    let flag = start_flag.clone();
+
+    let block_observer = BlockObserver {
+        context: ctx.clone(),
+        bitcoin_blocks: testing::btc::new_zmq_block_hash_stream(BITCOIN_CORE_ZMQ_ENDPOINT).await,
+    };
+
+    tokio::spawn(async move {
+        flag.store(true, Ordering::Relaxed);
+        block_observer.run().await
+    });
+
+    while !start_flag.load(Ordering::SeqCst) {
+        tokio::time::sleep(Duration::from_millis(10)).await;
+    }
+
+    // Let's do a sanity check that we do not have any donation or deposit.
+    assert!(fetch_output(&db, TxOutputType::Donation).await.is_empty());
+    assert!(fetch_input(&db, TxPrevoutType::Deposit).await.is_empty());
+
+    // ** Step 3 **
+    //
+    // Create a coinbase tx sending funds to the signers. In this case there is
+    // only one signer in the signer set.
+    let signer = Recipient::new(AddressType::P2tr);
+
+    // We need to have run DKG in order for the block observer to know
+    // which addresses to filter on.
+    let mut shares: EncryptedDkgShares = Faker.fake_with_rng(&mut rng);
+    shares.aggregate_key = signer.keypair.public_key().into();
+    shares.script_pubkey = shares.aggregate_key.signers_script_pubkey().into();
+    shares.dkg_shares_status = model::DkgSharesStatus::Verified;
+    db.write_encrypted_dkg_shares(&shares).await.unwrap();
+
+    // Okay, now to make the actual coinbase donation. We send some funds to
+    // their address.
+    let script_pub_key = shares.script_pubkey.deref();
+    let address = Address::from_script(script_pub_key, bitcoin::Network::Regtest).unwrap();
+
+    // We include also another donation to ensure we correctly process the block
+    let donation_amount = 123_456;
```

Comment:
nit: generating this amount with rng will make test a little bit better

Reply:
mmm I don't think in this case: we are not testing the parsing itself, we just need an amount to ensure we parsed the expected donation. Having a random amount here could add more noise.